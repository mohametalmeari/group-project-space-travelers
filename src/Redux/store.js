import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './features/missions/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
