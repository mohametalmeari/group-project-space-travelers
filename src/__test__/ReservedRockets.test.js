import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import ReservedRocketDisplay from '../__Mock_test/ReservedRockets';

describe('ReservedRocketDisplay Component', () => {
  test('renders reserved rockets', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Provider store={store}>
          <ReservedRocketDisplay />
        </Provider>
      </MemoryRouter>,
    );

    expect(getByText('My Rockets')).toBeInTheDocument();

    const reservedRocket = getByText(/reserved/i);
    expect(reservedRocket).toBeInTheDocument();

    expect(getByText('Currently no rockets reserved.')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <ReservedRocketDisplay />
        </Provider>
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
