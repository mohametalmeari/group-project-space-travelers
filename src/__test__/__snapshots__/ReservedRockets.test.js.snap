// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedRocketDisplay Component matches snapshot 1`] = `
<div>
  <div
    class="container"
    style="width: 60%;"
  >
    <h3>
      My Rockets
    </h3>
    <div>
      <div
        class="colums"
      >
        <p>
          Currently no rockets reserved.
        </p>
      </div>
    </div>
  </div>
</div>
`;
